src/s2_cells/mod.rs

```diff
@@ -50,3 +63,157 @@ impl S2Meta {
         meta
     }
 }
+
+#[derive(Copy, Clone)]
+pub struct S2CellMeta {
+    pub num_points: u64,
+}
+
+/// Just a wrapper that implements Display
+/// TODO(nnmm): Remove as soon as version 0.10 of s2 is released
+#[derive(Copy, Clone)]
+pub struct S2CellId(pub CellID);
+
+impl std::fmt::Display for S2CellId {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}", self.0.to_token())
+    }
+}
+
+impl PointCloud for S2Cells {
+    type Id = S2CellId;
+    type PointsIter = FilteredPointsIterator;
+
+    fn nodes_in_location(&self, query: &PointQuery) -> Vec<Self::Id> {
+        match &query.location {
+            PointLocation::AllPoints() => self.cells.keys().cloned().map(S2CellId).collect(),
+            PointLocation::Aabb(aabb) => {
+                self.cells_in_obb(&Obb::from(aabb), query.global_from_local.as_ref())
+            }
+            PointLocation::Obb(obb) => self.cells_in_obb(&obb, query.global_from_local.as_ref()),
+            PointLocation::Frustum(mat) => {
+                let world_from_clip = mat.inverse_transform().unwrap();
+                let points = CUBE_CORNERS
+                    .iter()
+                    .map(|p| Point3::from_homogeneous(world_from_clip * p));
+                self.cells_in_convex_hull(points)
+            }
+            PointLocation::OrientedBeam(_) => unimplemented!(),
+        }
+    }
+
+    fn encoding_for_node(&self, _: Self::Id) -> Encoding {
+        Encoding::Plain
+    }
+
+    fn points_in_node<'a>(
+        &'a self,
+        query: &PointQuery,
+        node_id: Self::Id,
+    ) -> Result<Self::PointsIter> {
+        let culling = query.get_point_culling();
+        let num_points = self.meta.cells[&node_id.0].num_points as usize;
+        let node_iterator = NodeIterator::from_data_provider(
+            &*self.data_provider,
+            self.encoding_for_node(node_id),
+            &node_id,
+            num_points,
+        )?;
+        Ok(FilteredPointsIterator {
+            culling,
+            node_iterator,
+        })
+    }
+}
+
+impl S2Cells {
+    /// Wrapper arround cells_in_convex_hull for Obbs
+    fn cells_in_obb(
+        &self,
+        obb: &Obb<f64>,
+        global_from_local: Option<&Isometry3<f64>>,
+    ) -> Vec<S2CellId> {
+        let obb = match global_from_local {
```

Comment:
Nit: Instead of returning a `Cow` of `obb`, can we just return the `obb.corners` directly as `points`, since the `obb` isn't used elsewhere?

Reply:
`creates a temporary which is freed while still in use` – unless we just clone.